Create install_arch.sh

CooperGerman opened the pull request:
I had this script working on a endeavourOS machine.

Donkie commented:
Thank you! I see a lot of similarities between this and the debian one, perhaps the script can be split up into several scripts and then having common parts be located in some `scripts/functions/` folder or something?

CooperGerman commented:
Hi, yes you are right i think some common stuff can be factorized into something as mentioned. I didn't want to take the initiative of this but i can give it a try if you want.

Donkie commented:
Yes please do :)

CooperGerman commented:
It seems like the requirements.txt file needed for `install_debian.sh` is generated outside of the script itself, is this intended or can i add the  
```sh
pip install pdm
pdm export -o requirements.txt --without-hashes > requirements.txt
```
lines to the generic installer ?

Donkie commented:
requirements.txt is generated in the CI job (to the root dir) to prevent people from needing to install pdm, which has proved to have quite annoying dependencies on some systems. So that cannot really be changed.

CooperGerman commented:
OK, i understand. What is the better way for me to test the installation process in this case ? can i emulate a release in order to reuse it for an installation ?

CooperGerman commented:
Hi, i updated the script and updated the README accordingly. 
I only tested it on arch for now. Pre-commit was OK.

Is there a way to ensure it works without me having to setup a VM ?

Donkie commented on scripts/install.sh:
```diff
@@ -51,38 +51,63 @@ else
+        echo -e "${ORANGE}Your operating system is not supported. Either try to install manually or reach out to spoolman github for support.${NC}"
+        exit 1
+    fi
+fi
```
If /etc/os-release is not available it should also fail

Donkie commented:
> Hi, i updated the script and updated the README accordingly. I only tested it on arch for now. Pre-commit was OK.
> 
> Is there a way to ensure it works without me having to setup a VM ?

I can test it for you on debian

Donkie commented on scripts/install.sh:
```diff
@@ -51,38 +51,63 @@ else
+if [[ -f /etc/os-release ]]; then
+    source /etc/os-release
+    if [[ "$ID_LIKE" == *"debian"* ]]; then
+        pkg_manager="apt-get"
```
Could be nice to add an echo here saying we detected Debian-based OS, same for arch

CooperGerman replied:
Done ;)

Donkie commented on scripts/install.sh:
```diff
@@ -51,38 +51,65 @@ else
-sudo apt-get update || exit 1
+if [[ -f /etc/os-release ]]; then
+    source /etc/os-release
+    if [[ "$ID_LIKE" == *"debian"* ]]; then
```
Turns out on Debian itself, ID_LIKE doesn't exist, only ID does, which is "debian". So you should check for either ID_LIKE like you do here, or `"$ID" == *"debian"*`

![image](https://github.com/Donkie/Spoolman/assets/2332094/79770192-17aa-4e44-be47-1b59a00f614a)

CooperGerman replied:
I check for both ID and ID_LIKE variables, this should do the trick for you.

Donkie commented on scripts/install.sh:
```diff
@@ -51,38 +51,65 @@ else
     echo -e "${ORANGE}Python pip is not installed. Installing pip...${NC}"
-    install_packages=1
+    if [[ "$pkg_manager" == "apt-get" ]]; then
+        packages+="python3-pip"
```
You need to prepend all of these strings with a blank space, otherwise the final packages string will be like `"python3-venvpython3-piplibpq-dev"` which of course apt-get doesn't understand. Not sure how pacman handles this

CooperGerman replied:
Of course, strange that i didn't trigger any issue with this one...

Donkie commented:
Looks good now and seems to work fine, thank you!
